### ISSUE TITLE
CC does not refresh dashboard_url following completion of async service provisionning

### ISSUE THREAD

**Issue by gberche-orange** [cloudfoundry/cloud_controller_ng#1390]
## Issue

CC does not support service brokers that provide an updated broker url upon completion of the async service instance provisioning. 

## Context

Some service brokers can't provide a dashboard URL in the synchronous service provisionning response, and rather return them in the `GET /v2/service_instances/:instance_id` endpoint. See related https://github.com/openservicebrokerapi/servicebroker/issues/498#issuecomment-411434165

## Steps to Reproduce

Steps run on PWS: 

```sh
git clone https://github.com/mattmcneeney/overview-broker.git
cf push -f examples/cloudfoundry/manifest.yaml --hostname overview-gberche
cf create-service-broker --space-scoped overview-broker admin password https://overview-gberche.cfapps.io
cf cs overview-service small overview-gberche
```

Check that catalog indeed specifies `"instances_retrievable": true`

```
curl -H "x-broker-api-version: 2.14" -u admin:password https://overview-gberche.cfapps.io/v2/catalog | jq
```

Use browser at https://overview-gberche.cfapps.io/dashboard to check calls made by CC to `GET /v2/service_instances/:instance_id`. None are made. CC isn't fetching any updated dashboard url

## Expected result

CC should call the `GET /v2/service_instances/:instance_id` upon successful completion of async service instance provisioning (last operation endpoint), and update the dashboard url stored in CC db, and displayed in users in CLI querying `GET /v2/spaces/3518c030-0b99-4075-834e-1b47e6ec7909/service_instances?q=name%3...`

## Current result

CC does not call the `GET /v2/service_instances/:instance_id` upon successful completion of async service instance provisioning 

/CC @mattmcneeney

**Comment by gberche-orange** [comment]
Hi @AartiKriplani 

I've added this pain to the [V3 Managed & User-Provided Service Instances Proposal](https://docs.google.com/document/d/1P5laZqVYjtl_boSdpksT5LHOFjhm6CQLzWVx24UpXpM)

I notice the current [V3 service instance endpoint documentation](http://v3-apidocs.cloudfoundry.org/version/3.76.0/index.html#the-service-instance-object) does not mention a dashboard url unlike [V2 endpoint](https://apidocs.cloudfoundry.org/12.0.0/service_instances/retrieve_a_particular_service_instance.html), and I guess should still be considered as experimental, even though it is not mentionned as such.

In which backlog service related service instance CC API V3 work can be tracked ?
* could not yet spot them in [VAT backlog](https://www.pivotaltracker.com/n/projects/2196383)
* the SAPI team as a related [V3 epic](https://www.pivotaltracker.com/epic/show/4129144) but I could not yet spot associated service instance story

**Comment by gberche-orange** [comment]
Re @addytripathi 

> Reopen this issue if you have a comments?

I wish this issue could remain open. Github does not allow non project members to reopen closed issues. I'd like to avoid submitting a duplicate issue to continue this conversation around inputs for V3 service endpoint.

**Comment by addytripathi** [comment]
Hi @gberche-orange .. sorry about the delay in responding to you. I'll reopen this issue. 

For reference, we're using this tracker for services API endpoints: https://www.pivotaltracker.com/n/projects/2105761

We're likely take up this work when we get to the service-instance resource.

**Comment by gberche-orange** [comment]
thanks @addytripathi I'll be watching the V3 [service-instances](https://www.pivotaltracker.com/n/projects/2105761/search?q=label%3A%22service-instances%22) label for related work.

**Comment by FelisiaM** [comment]
Hi @gberche-orange 

Thanks for raising this and apologies for the time it took to get to it.

We have just implemented the feature to make sure that if the dashboard URL is provided in the service instance fetch endpoint we update it at the end of a create and update async operation. This should be available in the next CAPI release.

I will close the issue now. Let us know if you have any further feedback about this.

Thanks,
Felisia

**Comment by gberche-orange** [comment]
Awesome, thanks @FelisiaM !

**Comment by elenasharma** [comment]
The fix for this issue was released in [capi-release 1.106.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.106.0). Thank you for raising this issue!
